python_modules/dagster/dagster/core/execution.py

```diff
@@ -508,6 +510,59 @@ def __init__(self, pipeline, errors, config_value, *args, **kwargs):
         super(PipelineConfigEvaluationError, self).__init__(error_msg, *args, **kwargs)
 
 
+def execute_externalized_plan(
+    pipeline,
+    step_keys,
+    inputs_to_marshal=None,
+    outputs_to_marshal=None,
+    environment=None,
+    reentrant_info=None,
+):
+    check.inst_param(pipeline, 'pipeline', PipelineDefinition)
+    check.list_param(step_keys, 'step_keys', of_type=str)
+    check_opt_two_dim_str_dict(inputs_to_marshal, 'inputs_to_marshal', value_type=str)
+    check.opt_dict_param(outputs_to_marshal, 'outputs_to_marshal', key_type=str, value_type=list)
+    check.opt_dict_param(environment, 'environment')
+    check.opt_inst_param(reentrant_info, 'reentrant_info', ReentrantInfo)
+
+    execution_plan = create_execution_plan(pipeline, environment)
+
+    inputs = defaultdict(dict)
+
+    for step_key, input_dict in inputs_to_marshal.items():
+        for input_name, file_path in input_dict.items():
+            step = execution_plan.get_step_by_key(step_key)
+            step_input = step.step_input_dict[input_name]
+            input_type = step_input.runtime_type
+
+            check.invariant(input_type.marshalling_strategy)
+
+            input_value = input_type.marshalling_strategy.unmarshal_value(file_path)
+            inputs[step_key][input_name] = input_value
+
+    results = execute_plan(
+        pipeline,
+        execution_plan,
+        environment=environment,
+        subset_info=ExecutionPlanSubsetInfo(included_steps=step_keys, inputs=inputs),
+        reentrant_info=reentrant_info,
+    )
+
+    for result in results:
+        if not (result.success and result.step.key in outputs_to_marshal):
```

Comment:
My only question about this is whether we want to error in some way if we don't see `result.success`.

Reply:
yup good call. i'll add an issue to track